### PR TITLE
Finalize v0.4.0

### PR DESCRIPTION
**Pull request by StandingPadAnimations**

### CODE REVIEW

**Comment by StandingPadAnimations** [comment]
@TheDuckCow Ready for review. It's mostly just some docs, the `v0.4.0` tag, and updating the version

File: CONTRIBUTING.md
Comment (TheDuckCow) [inline CONTRIBUTING.md:1]: I'm still running into an issue where when I run "poetry run python -m unittest test.tests" I get this error output, I'm guessing there's an extra step we need to put in this contributing guide which is the proper way to install dependencies? Pretend I know knowing about poetry, which is basically still true, so it's easy for any developer to get started with no gaps :) for bpy-build, I don't have a dedicate pyenv set up, as I think you mentions poetry means to handle that

```
bpy-build $ poetry run python -m unittest test.tests
E
======================================================================
ERROR: tests (unittest.loader._FailedTest.tests)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/unittest/loader.py", line 137, in loadTestsFromName
    module = __import__(module_name)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/patrickcrawford/Documents/gits/bpy-build/test/tests.py", line 33, in <module>
    import bpy_addon_build as bab
  File "/Users/patrickcrawford/Documents/gits/bpy-build/bpy_addon_build/__init__.py", line 31, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'


----------------------------------------------------------------------
Ran 1 test in 0.000s
```

This was after pulling the latest and running `poetry build  ` and `poetry run mypy --pretty bpy_addon_build` if that matters, but I guess those are separate/more informational.

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Getting Started
```
Comment (TheDuckCow) [inline docs/getting-started.md:1]: Would be good to clarify what this is getting started *for*

Maybe something like:

```suggestion
# Getting Started: Building your own addoon

To get started using bpy-build to manage building and reloading your own blender addon project, follow the steps below.
```

File: docs/tests.md
Comment (TheDuckCow) [inline docs/tests.md:1]: I guess I realize I'm a bit confused about this - is this implying that e.g. in MCprep, we should be having the unit tests runner itself, as an execution of one of the test suite classes, perform the duty of installing the addon? Not sure that makes the most sense - especially when we consider that we want to be able to specify specific tests to run if we're focussing on one at a time to fix/implement (something I do all the time) - if the builder itself is a test, you'd always have to specify that one in order to put the new code in place. But maybe I misunderstand the idea here.

**Review by TheDuckCow — COMMENTED** [review]
Leaving comments first to make sure I can get it working right based on the docs provided

File: CONTRIBUTING.md
Comment (StandingPadAnimations) [reply CONTRIBUTING.md:1]: Did you run `poetry install`? That's the command that installs the dependencies in the created virtual environment.

File: docs/tests.md
Comment (StandingPadAnimations) [reply docs/tests.md:1]: Yeah, I can see the confusion. I'll remove this for now, and add it back at a later date (the next release is mostly focusing on tying up loose ends, improving error handling, etc.). The idea is to call the main function directly in unittests over using a subprocess, although the given example isn't the best. 

Ideally, `main` could be called in [`setUpClass`](https://docs.python.org/3/library/unittest.html#setupclass-and-teardownclass), but we can't “mock” `sys.argv` in `setUpClass`. However, `argparse` [supports parsing a supplied list](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args) other than `sys.argv`, so we could add an optional `args` argument to `main` in `v0.4.1`

**Comment by StandingPadAnimations** [comment]
@TheDuckCow when do you think we can get this merged?

**Review by TheDuckCow — APPROVED** [review]
The poetry install step was the one I was missing, everything builds and all tests pass now. Let's go ahead and merge and cut the release

**Comment by StandingPadAnimations** [comment]
BpyBuild 0.4 has been released on PyPi

**Pull request merged**